addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -147,6 +162,26 @@ type ObjectReference struct {
 	Name string `json:"name"`
 }
 
+// ConfigReference is a reference to the current add-on configuration.
+// This resource is used to locate the configuration resource for the current add-on.
+type ConfigReference struct {
+	// This field is synced from ClusterManagementAddOn configGroupResource field.
+	ConfigGroupResource `json:",inline"`
+
+	// This field is synced from ClusterManagementAddOn defaultConfig and ManagedClusterAddOn config fields.
+	// If both of them are defined, the ManagedClusterAddOn config will overwrite the ClusterManagementAddOn
+	// defaultConfig.
+	ConfigReferent `json:",inline"`
+
+	// version of the add-on configuration.
```

Comment:
When the controller sets this value, how does it determine what the version is?  Is it an API version or more like a revision of the configuration like a generation.

Reply:
it's the api version, the addon framework use rest mapper to discovery the config api version with the config resource and group that are defined in the clustermanagementaddon, and add the version to managedclusteraddon status.

For an addon implementation, it can get the config full meta info (group, resource, version and name/namespace) from its managedclusteraddon, then the implementation use them to get the config cr, and render its addon agent

it will be easy to get the addon config cr for addon implementation with the config api version

Reply:
UPDATE

Remove the version from the managedclusteraddon status, for addon implementation, it should known the exact apiversion of the config, so the config apiversion is not necessary, and we will ask implementation to provide the config gvr to the addon framework, thus the framework will have an exact type to handle